frontend/src/components/print/print-react/components/picasso/PicassoPage.jsx

```diff
@@ -0,0 +1,122 @@
+// eslint-disable-next-line no-unused-vars
+import React from 'react'
+import { Page, Text, View } from '@react-pdf/renderer'
+import styles from '../styles.js'
+import picassoStyles from './picassoStyles.js'
+import TimeColumn from './TimeColumn.jsx'
+import DayColumn from './DayColumn.jsx'
+import TimeColumnSpacer from './TimeColumnSpacer.jsx'
+import DayHeader from './DayHeader.jsx'
+import PicassoFooter from './PicassoFooter.jsx'
+import YSLogo from './YSLogo.jsx'
+import Categories from './Categories.jsx'
+import dayjs from '@/common/helpers/dayjs.js'
+import { calculateBedtime } from '@/common/helpers/picasso.js'
+
+/**
+ * Generates an array of time row descriptions, used for labeling the vertical axis of the picasso.
+ * Format of each array element: [hour, weight] where weight determines how tall the time row is rendered.
+ *
+ * @param getUpTime the first time row (the very top of the picasso)
+ * @param bedtime the last time row (the very bottom of the picasso)
+ * @param timeStep size of the time rows, in hours
+ * @returns {*[[hour: number, weight: number]]}
+ */
+function generateTimes({ getUpTime, bedtime, timeStep }) {
+  const times = []
+  for (let i = 0; getUpTime + i * timeStep < bedtime; i++) {
+    // TODO The weight could also be generated depending on the schedule entries present in the camp:
+    //   e.g. give less weight to hours that contain no schedule entries.
+    const weight = 1
+    times.push([getUpTime + i * timeStep, weight])
+  }
+  // this last hour is only needed for defining the length of the day. The weight should be 0.
+  times.push([bedtime, 0])
```

Comment:
Yes, this is still needed. The complete times array gets passed into DayColumn, where it is used to calculate the vertical positioning of the schedule entries (e.g. in the `percentage` and `dayBoundariesInMinutes` functions). Maybe this could be refactored, but I'm not yet sure the code would get simpler then, and I don't really see this as part of this PR.